src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -0,0 +1,608 @@
+#include "mongoBackend/MongoCommonSubscription.h"
+#include "mongoBackend/dbConstants.h"
+#include "mongoBackend/safeMongo.h"
+#include "mongoBackend/mongoSubCache.h"
+#include "common/defaultValues.h"
+
+#include "cache/subCache.h"
+
+#include "mongo/client/dbclient.h"
+
+using namespace mongo;
+using namespace ngsiv2;
+
+/* ****************************************************************************
+*
+* setExpiration -
+*
+*/
+static void setExpiration(const SubscriptionUpdate& subUp, const BSONObj& subOrig, BSONObjBuilder* b)
+{
+  if (subUp.expiresProvided)
+  {
+    setExpiration(subUp, b);
+  }
+  else
+  {
+    if (subOrig.hasField(CSUB_EXPIRATION))
+    {
+      long long expires = getLongFieldF(subOrig, CSUB_EXPIRATION);
+      b->append(CSUB_EXPIRATION, expires);
+      LM_T(LmtMongo, ("Subscription expiration: %lu", expires));
+    }
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setHttpInfo -
+*
+*/
+static void setHttpInfo(const SubscriptionUpdate& subUp, const BSONObj& subOrig, BSONObjBuilder* b)
+{
+  if (subUp.notificationProvided)
+  {
+    setHttpInfo(subUp, b);
+  }
+  else
+  {
+    // reference is a mandatory field and extended has a clear mapping to false in the case of missing field
+    std::string reference = getStringFieldF(subOrig, CSUB_REFERENCE);
+    bool extended         = subOrig.hasField(CSUB_EXTENDED) ? getBoolFieldF(subOrig, CSUB_EXTENDED) : false;
+
+    b->append(CSUB_REFERENCE, reference);
+    b->append(CSUB_EXTENDED,  extended);
+
+    LM_T(LmtMongo, ("Subscription reference: %s", reference.c_str()));
+    LM_T(LmtMongo, ("Subscription extended: %s", extended? "true" : "false"));
+
+    if (subOrig.hasField(CSUB_METHOD))
+    {
+      std::string method = getStringFieldF(subOrig, CSUB_METHOD);
+      b->append(CSUB_METHOD, method);
+      LM_T(LmtMongo, ("Subscription method: %s", method.c_str()));
+    }
+
+    if (subOrig.hasField(CSUB_HEADERS))
+    {
+      BSONObj headers = getObjectFieldF(subOrig, CSUB_HEADERS);
+      b->append(CSUB_HEADERS, headers);
+      LM_T(LmtMongo, ("Subscription headers: %s", headers.toString().c_str()));
+    }
+
+    if (subOrig.hasField(CSUB_QS))
+    {
+      BSONObj qs = getObjectFieldF(subOrig, CSUB_QS);
+      b->append(CSUB_QS, qs);
+      LM_T(LmtMongo, ("Subscription qs: %s", qs.toString().c_str()));
+    }
+
+    if (subOrig.hasField(CSUB_PAYLOAD))
+    {
+      std::string payload = getStringFieldF(subOrig, CSUB_PAYLOAD);
+      b->append(CSUB_PAYLOAD, payload);
+      LM_T(LmtMongo, ("Subscription payload: %s", payload.c_str()));
+    }
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setThrottling -
+*
+*/
+static void setThrottling(const SubscriptionUpdate& subUp, const BSONObj& subOrig, BSONObjBuilder* b)
+{
+  if (subUp.throttlingProvided)
+  {
+    setThrottling(subUp, b);
+  }
+  else
+  {
+    if (subOrig.hasField(CSUB_THROTTLING))
```

Comment:
I prefer current way, to have the structure of all set*() functions in this file homogeneous. See for example a case the suggested strategy cannot be used: setHttpInfo().